config/merge.go

```diff
@@ -25,14 +27,21 @@ var (
 	}
 )
 
-// CreateConfig unmarshals bytes to config and creates config based on version
+// CreateConfig unmarshals bytes of a YAML manifest file and returns a new
+// Config. Initialize any defaults that can't be parsed (but are optional)
+// across various file formats. Most of these can remain unused.
+//
+// This function only handles parsing YAML in the general case. Any other file
+// format validation should be handled by the caller.
 func CreateConfig(bytes []byte) (*Config, error) {
 	var config Config
 	if err := yaml.Unmarshal(bytes, &config); err != nil {
 		return nil, err
 	}
 
-	if config.Version != "2" {
+	parts := strings.Split(config.Version, ".")
+	major, _ := strconv.Atoi(parts[0])
```

Comment:
We shouldn't skip the error here

Reply:
Normally I'm 100% with you, and please correct me if I'm wrong, but I'm not sure what error we'd capture here or would want to bubble up considering that we're guaranteed to have `0` output by [`strconv.Atoi`](https://github.com/golang/go/blob/46aa9f5437b000fad3696b0cd9fd97995da16411/src/strconv/atoi.go#L202) regardless of input or error. We should always handle a `0` versioned compose file since they're the default in all cases.

Reply:
@vdemeester, interested in your thoughts here. Thanks.

Reply:
@cheapRoc `version` is of type string (in the schema), so I could put `a.b` it would pass the validation. That's why I think we should check for errors.

Reply:
I agree, we should validate the full version number since it's a string. Maybe that's something that needs to be handled before this spot (?). That's not what I was trying to achieve here.

At this line, and the other you pointed out, I'm only parsing a single integer value. The `a` of `a.b`, and I should be handling the default case as `0` appropriately.

Reply:
That’s what I’ll do, I’ll formally parse the schema version major and minor and try to reuse it in other places.